app/controllers/katello/api/v2/repository_sets_controller.rb

```diff
@@ -14,7 +14,7 @@ class Api::V2::RepositorySetsController < Api::V2::ApiController
     end
 
     api :GET, "/products/:product_id/repository_sets", N_("List repository sets for a product.")
-    param :product_id, :number, :required => true, :desc => N_("ID of a product to list repository sets from")
+    param :product_id, :number, :required => false, :desc => N_("ID of a product to list repository sets from")
```

Comment:
You could remove the `required` part altogether (default is false).

Reply:
I'd rather go with uniformity on this one (see all the required => false throughout)

Reply:
@cfouant This change still needs to be reverted.